features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/AdminPages/Addons/InputListBase.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { createMuiTheme } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+
+let id;
+
+const theme = createMuiTheme();
+theme.spacing(2);
+
+const InputList = (props) => {
+    const {
+        onInputListChange, initialList, inputLabelPrefix, helperText, addButtonLabel,
+    } = props;
+    const [userInputItems, setUserInputItems] = useState([]);
+
+    const handleInput = (e) => {
+        let tempItems = userInputItems.filter((item) => item.key !== e.target.name);
+        tempItems = [...tempItems, { key: '' + e.target.name, value: e.target.value }];
+        tempItems.sort((a, b) => {
+            return a.key - b.key;
+        });
+        setUserInputItems(tempItems);
+    };
+
+    const handleDelete = (deletingKey) => {
+        const tempItems = userInputItems.filter((item) => (item.key !== deletingKey));
+        setUserInputItems(tempItems);
+    };
+
+    useEffect(() => {
+        const nonEmptyItems = [];
+        for (let i = 0; i < userInputItems.length; i++) {
+            if (userInputItems[i].value && userInputItems[i].value.trim() !== '') {
+                nonEmptyItems.push(userInputItems[i].value.trim());
+            }
+        }
+        onInputListChange(nonEmptyItems);
+    }, [userInputItems]);
+
+    useEffect(() => {
+        id = 1;
```

Comment:
Have we ignored eslint check in this file ? , AFAIR this is an eslint rule violation

Reply:
eslint check is enabled for this file.